pkg/interface/publish/src/js/components/lib/join.js

```diff
@@ -1,13 +1,125 @@
 import React, { Component } from 'react'
+import classnames from 'classnames';
+import { Route, Link } from 'react-router-dom';
+import urbitOb from 'urbit-ob';
 
 //TODO textarea + join button to make an api call
 export class JoinScreen extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      book: '/',
+      error: false
+    };
+
+    this.bookChange = this.bookChange.bind(this);
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+      const { props, state } = this;
+      if (state.book in props.notebooks) {
+        props.history.push('/~publish');
+      }
+   }
+
+  onClickJoin() {
+    const { props, state } = this;
+
+    let text = state.book;
+    // an error condition to prevent double joins?
+    if (text in props.notebooks ||
```

Comment:
Likewise, this error check will be set to `true` if I type `~zod`, or any other notebook owner I'm subscribed to.